operator.go

```diff
@@ -6,10 +6,21 @@ import (
 	"text/template"
 )
 
+type encoding string
```

Comment:
I somehow can explain this in words apparently. Let me explain in code. I would like to have this. 

```
type encoding string

const (
	UTF8 encoding = "utf-8"
	GzipBase64 encoding = "gzip+base64"
)
```

- We do not need a string method. 
- We do not need to use empty strings to get some default value. 
- We use the private type to protect our choices. 
- We provide all available choices via constants. 

The constant has only to be applied to the template params, which you would also do with your current encoding type. I only think what I suggest is more strict and more straight forward. At the end, do what you think is right.